src/source/Common/AwsV4Signer.c

```diff
@@ -73,7 +73,7 @@ STATUS generateAwsSigV4Signature(PRequestInfo pRequestInfo, PCHAR dateTimeStr, B
     hmacSize = SIZEOF(hmac);
     CHK_STATUS(generateRequestHmac((PBYTE) pScratchBuf, curSize, (PBYTE) dateTimeStr, SIGNATURE_DATE_STRING_LEN * SIZEOF(CHAR), hmac, &hmacSize));
     CHK_STATUS(generateRequestHmac(hmac, hmacSize, (PBYTE) pRequestInfo->region, (UINT32) STRLEN(pRequestInfo->region), hmac, &hmacSize));
-    CHK_STATUS(generateRequestHmac(hmac, hmacSize, (PBYTE) KINESIS_VIDEO_SERVICE_NAME, (UINT32) STRLEN(KINESIS_VIDEO_SERVICE_NAME), hmac, &hmacSize));
+    CHK_STATUS(generateRequestHmac(hmac, hmacSize, (PBYTE) pRequestInfo->service, (UINT32) STRLEN(pRequestInfo->service), hmac, &hmacSize));

```

Comment:
STRLEN is a very dangerous API. In the previous case, it should not have been used but instead used ARRAY_SIZE. Either way, it was hard-coded value which was kind of OK. In this case, the size should be calculated by calling STRNLEN - with max size of MAX_SERVICE_NAME_LEN. STRNLEN should also be used in the lines above and below.